common/AppRouter.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { HashRouter, BrowserRouter, Route, Switch } from 'react-router-dom';
+
+import { ErrorScreen, PalettePage, LogOutPrompt } from 'components';
+import {
+  BroadcastTx,
+  CheckTransaction,
+  Contracts,
+  ENS,
+  GenerateWallet,
+  SendTransaction,
+  SignAndVerifyMessage,
+  SupportPage
+} from 'containers';
+import { Layout } from 'v2/features/Layout';
+import { Home, PageNotFound } from 'v2/features';
+import { useDevMode } from 'v2/services';
+import { ScreenLockProvider } from 'v2/providers';
+import { IS_DEV, IS_PROD, IS_DOWNLOADABLE } from 'v2/utils';
+import { ROUTE_PATHS } from 'v2/config';
+import {
+  APP_ROUTES,
+  PageVisitsAnalytics,
+  LegacyRoutesHandler,
+  HomepageChoiceRedirect,
+  PrivateRoute
+} from 'v2/routing';
+
+interface AppRouterProps {
+  error?: Error | undefined;
+}
+
+export const AppRouter = (props: AppRouterProps) => {
+  const { isDevelopmentMode } = useDevMode();
+  const { error } = props;
+
+  if (error !== undefined && !isDevelopmentMode) {
+    return <ErrorScreen error={error} />;
+  }
+
+  const Router: any = IS_DOWNLOADABLE && IS_PROD ? HashRouter : BrowserRouter;
+
+  return (
+    <Router>
+      <ScreenLockProvider>
+        <PageVisitsAnalytics>
+          <HomepageChoiceRedirect>
+            <Switch>
+              {/* To avoid fiddling with layout we provide a complete route to home */}
+              <Route path={ROUTE_PATHS.ROOT.path} component={Home} exact={true} />
+              <Route path={ROUTE_PATHS.HOME.path} component={Home} exact={true} />
+              <Route path="/account" component={SendTransaction} exact={true} />
+              <Route path="/generate" component={GenerateWallet} />
+              <Route path="/contracts" component={Contracts} />
+              <Route path="/ens" component={ENS} exact={true} />
+              <Route path="/sign-and-verify-message" component={SignAndVerifyMessage} />
+              <Route path="/tx-status" component={CheckTransaction} exact={true} />
+              <Route path="/pushTx" component={BroadcastTx} />
+              <Route path="/support-us" component={SupportPage} exact={true} />
+              <Layout>
+                <Switch>
+                  {APP_ROUTES.filter(route => !route.seperateLayout).map((config, idx) => (
+                    <PrivateRoute key={idx} {...config} />
+                  ))}
+                  <Route component={PageNotFound} />
+                </Switch>
+              </Layout>
+              {IS_DEV && <Route path="/dev/palette" component={PalettePage} exact={true} />}
```

Comment:
Ah true, the default Route takes precedence!
I choose to remove the Route, and but leave the component in the folder. To avoid touching anything in v1.